backend/src/main/resources/db/migration/V10__dashboard_indexes.sql

```
@@ -0,0 +1,42 @@
+--      * Queries we write — findByProduct (product detail, the single-product trend, and the
+--        deprecated list endpoint, which calls it once PER PRODUCT on a page), plus this issue's
+--        countByProduct for the listing cap and findByProductIdIn for the page's sparklines.
+--      * The database itself, on DELETE. The FK has no ON DELETE clause, so it defaults to NO ACTION:
+--        before removing a product row, Postgres must prove no tracked_item still references it. That
+--        check is a scan of the child table per deleted parent — the classic unindexed-FK trap, and
+--        the one nobody profiles because it hides inside a DELETE.
+--
+--    Honest scale note: at the admission cap (~500 products x 20 listings) tracked_item is a few
+--    hundred pages and a sequential scan of it is sub-millisecond from cache. This is cheap
+--    insurance that stops the cost scaling with the table, not a fix for measured pain.
+CREATE INDEX idx_tracked_item_product ON tracked_item (product_id);
```

SQLFluff (4.2.2)

[error] CREATE INDEX should use CONCURRENTLY to avoid locking the table during the build.

(PG01)